up

sambacha opened the pull request:
Fixes # .

### Motivation
<!--
Does this solve a bug? Enable a new use-case? Improve an existing behavior? Concrete examples are helpful here.
-->

### Solution
<!--
What is the solution here from a high level. What are the key technical decisions and why were they made?
-->

### Open questions
<!--
(optional) Any open questions or feedback on design desired?
-->

patrick-ogrady committed:
Add address derivation

patrick-ogrady committed:
prepare_metadata functions

patrick-ogrady committed:
Add create_payloads

patrick-ogrady committed:
Combine signatures support

patrick-ogrady committed:
Add support for parse

patrick-ogrady committed:
Add support for transaction hash

patrick-ogrady committed:
nits

patrick-ogrady committed:
Update endpoints

patrick-ogrady committed:
Add flows to README

patrick-ogrady committed:
Update README with more explanation and context

patrick-ogrady committed:
Update api.yaml

patrick-ogrady committed:
Add docs for new models

patrick-ogrady committed:
Add notes about signature schemes

patrick-ogrady committed:
Add crypto formats

patrick-ogrady committed:
Fix double space issue

patrick-ogrady committed:
Add supported curvetypes and signaturetypes to README

patrick-ogrady committed:
run make gen

patrick-ogrady committed:
nits

patrick-ogrady committed:
Merge pull request #21 from coinbase/patrick/construction-api

Add Construction API (formerly Wallet API)

patrick-ogrady committed:
fix misspelling in operation schema (#6)

patrick-ogrady committed:
Reformat CurveType and SignatureType

patrick-ogrady committed:
Add oldest_block_identifier to NetworkStatusResponse (#7)

patrick-ogrady committed:
Remove MempoolRequest (#8)

patrick-ogrady committed:
Update /account/balance documentation (#9)

patrick-ogrady committed:
Add error details (#10)

patrick-ogrady committed:
Misspelling in transaction schema (#17)

patrick-ogrady committed:
Add historical_balance_lookup (#19)

patrick-ogrady committed:
Nits

patrick-ogrady committed:
build

patrick-ogrady committed:
Fix bool type

patrick-ogrady committed:
check validity in circle

patrick-ogrady committed:
Merge pull request #22 from coinbase/patrick/misc-issues

Fix Outstanding Issues for v1.4.0 Launch

patrick-ogrady committed:
Update links after website update

patrick-ogrady committed:
Merge pull request #24 from coinbase/patrick/update-readme-links

Update Website Links

matheusd committed:
Improve Error description

This improves the description of the Error model by indicating that
error messages must be unique and cannot have contextual information.

patrick-ogrady committed:
Merge pull request #25 from matheusd/error

Define Error.Code as authoritative

matheusd committed:
Add /block idempotency requirement

This adds a requirement that all calls to the /block endpoint be
idempotent when identifying a block by hash.

This ensures blocks are inalterable and clients can safely assume that
once they've received a specific block from a Rosetta implementation
that block won't change.

patrick-ogrady committed:
Merge pull request #26 from matheusd/inalterable-blocks

Add /block idempotency requirement

patrick-ogrady committed:
Unify transaction identifier response

patrick-ogrady committed:
Update default status code to 500 for errors

patrick-ogrady committed:
Make gen

patrick-ogrady committed:
Merge pull request #29 from coinbase/patrick/unify-tx-identifier

TransactionIdentifierResponse + change default status to 500

patrick-ogrady committed:
Add coin construct

patrick-ogrady committed:
Add CoinChange

patrick-ogrady committed:
Add descriptions

patrick-ogrady committed:
build

patrick-ogrady committed:
Merge pull request #30 from coinbase/patrick/sync-status

Add Coin Model

patrick-ogrady committed:
Add sync status

patrick-ogrady committed:
Make build

patrick-ogrady committed:
nits

patrick-ogrady committed:
Merge pull request #31 from coinbase/patrick/partial-sync-status

Support Pre-Query Status

AlanVerbner committed:
Improve descroption on `/block/transaction` regarding pagination

patrick-ogrady committed:
Merge pull request #32 from AlanVerbner/feature/block-transaction-pagination

Improve `/block/transaction` description regarding pagination

patrick-ogrady committed:
Add suggested fee to metadata response

patrick-ogrady committed:
Update version

patrick-ogrady committed:
Add suggested fee to preprocess request

patrick-ogrady committed:
Build

patrick-ogrady committed:
nits

patrick-ogrady committed:
Merge pull request #33 from coinbase/patrick/suggested-fee

Improved Fee Handling

patrick-ogrady committed:
First pass

patrick-ogrady committed:
Cleanup notes

patrick-ogrady committed:
Make build

patrick-ogrady committed:
Merge pull request #34 from coinbase/patrick/metadata-note

Add more details about Metadata population

patrick-ogrady committed:
Add block omitted field

patrick-ogrady committed:
Build

patrick-ogrady committed:
Better solution

patrick-ogrady committed:
Merge pull request #35 from coinbase/patrick/block-missing

Support Omitted Blocks

patrick-ogrady committed:
Fix RelatedOperations example

Commit:
add support for ecschnorr for zilliqa blockchain

Commit:
update api.json

patrick-ogrady committed:
Merge pull request #37 from coinbase/patrick/fix-related-operation-example

Fix RelatedOperations example

Commit:
rename ecschnorr to schnorr_1

Commit:
update README with link to Zilliqa's schnorr library implementation

patrick-ogrady committed:
Merge pull request #38 from teye/master

feat/add-ecschnorr-support-for-zilliqa

joeqian10 committed:
Add secp256r1 to CurveType

joeqian10 committed:
Add secp256r1 to CurveType

AlanVerbner committed:
fix: fixed Signers description to explicitly mention addresses are required

patrick-ogrady committed:
Merge pull request #41 from AlanVerbner/fix/signers-description

fix: fixed Signers description to explicitly mention addresses are required

joeqian10 committed:
modify api.json with signature

patrick-ogrady committed:
Merge pull request #39 from joeqian10/master

Add secp256r1 to CurveType

lk86 committed:
master Add tweedle CurveType

lk86 committed:
master Add schnorr_tweedle_fq signature scheme

lk86 committed:
master Decouple curve name from SignatureType

lk86 committed:
master Run make gen to update api.json

lk86 committed:
master Update api.json again

lk86 committed:
Revert "master Update api.json again"

This reverts commit 6f66b3c2793d9d7b6f8cf7c3212be82bed5a6047.

lk86 committed:
Revert "master Run make gen to update api.json"

This reverts commit 67488475a790106c6e3bf9b619520220959d1459.

lk86 committed:
Merge remote-tracking branch 'coinbase/master'

lk86 committed:
master Codegen for changes from master and coda changes

patrick-ogrady committed:
Add public_keys to preprocess and payloads

patrick-ogrady committed:
Make gen

patrick-ogrady committed:
nits

patrick-ogrady committed:
Merge pull request #45 from coinbase/patrick/preprocess-pubkeys

Access PublicKeys During Construction

lk86 committed:
master Add signature tweedle and schnorr_poseidon to README.md

patrick-ogrady committed:
Pass PublicKeys to ConstructionMetadata

patrick-ogrady committed:
make gen

patrick-ogrady committed:
nits

lk86 committed:
master Update signaturetype and README to include details on schnorr_poseidon challenge

patrick-ogrady committed:
Merge pull request #46 from coinbase/patrick/pubkeys-to-metadata

Provide PubKeys to /construction/metadata

lk86 committed:
master Update api.json from api.yaml

patrick-ogrady committed:
Merge pull request #44 from CodaProtocol/master

Support Coda with Tweedle curve and SchnorrPoseidon signatures

patrick-ogrady committed:
Add model changes

patrick-ogrady committed:
Build

patrick-ogrady committed:
Merge pull request #48 from coinbase/patrick/deprecate-address

Deprecate Address-based Fields

cindyxkuang committed:
updated types field description in Operation

cindyxkuang committed:
gen

cindyxkuang committed:
Merge pull request #51 from coinbase/cindy

updated types field description in Operation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/freight-trust/rosetta-specifications?pullRequest=2) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/freight-trust/rosetta-specifications?pullRequest=2) before we can accept your contribution.<br/>**0** out of **3** committers have signed the CLA.<br/><br/>:x: patrick-ogrady<br/>:x: cindyxkuang<br/>:x: KNWR<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/freight-trust/rosetta-specifications?pullRequest=2) it.</sub>

patrick-ogrady committed:
Create parity between ConstructionMetadataRequest and
ConstructionPreprocessResponse

patrick-ogrady committed:
build

patrick-ogrady committed:
Merge pull request #54 from coinbase/patrick/options-parity

[Construction] Options Parity

patrick-ogrady committed:
First pass at call

patrick-ogrady committed:
Update Allow.call_methods description

patrick-ogrady committed:
Write CallRequest and CallResponse description

patrick-ogrady committed:
Complete /call description

patrick-ogrady committed:
make gen

patrick-ogrady committed:
nits

patrick-ogrady committed:
Add modules section to README

patrick-ogrady committed:
Merge pull request #55 from coinbase/patrick/call-endpoint

/call endpoint

patrick-ogrady committed:
Add /call to online methods

patrick-ogrady committed:
Merge pull request #56 from coinbase/patrick/readme-nits

README Nits

patrick-ogrady committed:
First pass at adding BalanceExemption

patrick-ogrady committed:
Keep working

patrick-ogrady committed:
First pass on deets

patrick-ogrady committed:
Add license

patrick-ogrady committed:
Nits

patrick-ogrady committed:
Make gen

patrick-ogrady committed:
Nits

patrick-ogrady committed:
Merge pull request #57 from coinbase/patrick/support-exempt-balances

Support Account Balance Tracking "Exemptions"

KNWR committed:
Replace broken commit guide link

patrick-ogrady committed:
Merge pull request #58 from KNWR/kanwar/fix-broken-contributing-link

Replace broken link for commit guidelines

patrick-ogrady committed:
Add timestamp_start_index

patrick-ogrady committed:
gen

patrick-ogrady committed:
Merge pull request #60 from coinbase/patrick/timestamp-start-index

Add timestamp_start_index

patrick-ogrady committed:
Add Description to Error

patrick-ogrady committed:
gen

patrick-ogrady committed:
Merge pull request #61 from coinbase/patrick/error-description

Error Descriptions

patrick-ogrady committed:
Update README with reference implementations

patrick-ogrady committed:
nits

patrick-ogrady committed:
Merge pull request #62 from coinbase/patrick/cleanup-readme

Update README with reference implementations

patrick-ogrady committed:
remove status as required

patrick-ogrady committed:
update version

patrick-ogrady committed:
make gen

patrick-ogrady committed:
Merge pull request #63 from coinbase/patrick/optional-operation-status

Make `Operation.Status` optional

patrick-ogrady committed:
Add AccountCoinsResponse

patrick-ogrady committed:
Add mempool_coins to allow

patrick-ogrady committed:
Add note about not implementing /account/coins for account-based

patrick-ogrady committed:
make gen

patrick-ogrady committed:
Add additional docs

patrick-ogrady committed:
add more notes

patrick-ogrady committed:
Merge pull request #64 from coinbase/patrick/account-coins

Add /account/coins Endpoint

patrick-ogrady committed:
specify currencies to retrieve

patrick-ogrady committed:
make gen

patrick-ogrady committed:
Merge pull request #65 from coinbase/patrick/coin-lookup

/account/balance + /account/coins restricted queries

patrick-ogrady committed:
Add first indexer endpoints

patrick-ogrady committed:
Add scaffolding for events

patrick-ogrady committed:
Start populating event items

patrick-ogrady committed:
Add descriptions

patrick-ogrady committed:
Add description for search transactions

patrick-ogrady committed:
Add offset to query

patrick-ogrady committed:
make gen

patrick-ogrady committed:
Update README to comment on indexers

patrick-ogrady committed:
cleanup readme

patrick-ogrady committed:
fix spelling

patrick-ogrady committed:
nits

patrick-ogrady committed:
Merge pull request #66 from coinbase/patrick/indexer-enabled-endpoints

Indexer Endpoints

patrick-ogrady committed:
make operator optional

patrick-ogrady committed:
make gen

patrick-ogrady committed:
update version

patrick-ogrady committed:
make gen

patrick-ogrady committed:
Merge pull request #67 from coinbase/patrick/optional-operator

`SearchTransactionsRequest.Operator` Optional

patrick-ogrady committed:
update spec

patrick-ogrady committed:
make gen

patrick-ogrady committed:
Merge pull request #70 from coinbase/patrick/total-count

[indexer] Add `total_count` field

patrick-ogrady committed:
first pass

patrick-ogrady committed:
add synced field

patrick-ogrady committed:
make gen

patrick-ogrady committed:
nits

patrick-ogrady committed:
add more details about why current_index exists

patrick-ogrady committed:
nits

patrick-ogrady committed:
Merge pull request #71 from coinbase/patrick/network-status-tip

Return `Synced` Status

varunpcb committed:
Add Support for Related Transactions (#74)

* add modeling for related transactions

* bump version

* fix quotes

* nit

shrimalmadhur committed:
update readme with correct make function

shrimalmadhur committed:
update golang image for addlicense

nejoshi committed:
Merge pull request #81 from coinbase/madhur/readme-fix

update readme with correct validation syntax

rzheng committed:
Add description for hex-bytes in SigningPayload (#77)

matheusd committed:
Fix typo in SyncStatus (#75)

HenryYYang committed:
Add hash case sensitivity to the spec

Signed-off-by: Henry Yang <yhenry@gmail.com>

HenryYYang committed:
Merge pull request #83 from HenryYYang/add_hash_case

Add hash case sensitivity to the spec

racbc committed:
Update CONTRIBUTING.md file (#86)

* Update CONTRIBUTING.md file

* Post-review edits to contributing.md

* Post-review update of CONTRIBUTING.md file

racbc committed:
Fix typo in CONTRIBUTING.md (#90)

irisZhangCB committed:
Add pallas curve to spec (#93)

shrimalmadhur committed:
fix case issue (#94)

shrimalmadhur committed:
update version number (#95)

shrimalmadhur committed:
update circleci (#96)

racbc committed:
Update README and CONTRIBUTING files (#92)

* Update README and CONTRIBUTING files

These updates should facilitate contributions by developers.

* Edit README file formatting and text

* Update formatting in README file

* Update CONTRIBUTING file after feedback

* Post-peer updates to CONTRIBUTING.md file

* Post-review updates to README.md

* Edit formatting at top of README file

* Formatting update for README.md

racbc committed:
Update text in Issue Templates (#97)

racbc committed:
Fix rosetta-cli link to correct repo

racbc committed:
Merge pull request #98 from racbc/readme-link-fix

Fix rosetta-cli link to correct repo

irisZhangCB committed:
replace CircleCI with gh actions (#102)

* replace CircleCI with gh actions

* Update license script

racbc committed:
Remove peers as a required field in NetworkStatusResponse (#103)

racbc committed:
Update API version to 1.4.13 (#105)

jingweicb committed:
remove public key in metadata

jingweicb committed:
Merge pull request #106 from coinbase/jingwei/remove-public-keys

remove public key in metadata

jingweicb committed:
add public key back

jingweicb committed:
remove max_fee and mulltiplier in preprocess

jingweicb committed:
Merge pull request #109 from coinbase/jingwei/add-publickey-back

[metadata]add public key back

jingweicb committed:
updates

jingweicb committed:
Merge pull request #110 from coinbase/jingwei/remove-fields-preprocess

[preprocess]remove max_fee and mulltiplier in preprocess request

sleepdefic1t committed:
feat: support bip340 (#114)

xiaying-peng committed:
Rename to Mesh and upgrade Golang version. (#120)

xiaying-peng committed:
fix (#121)

xiaying-peng committed:
fix ref (#122)

rafalio-cb committed:
bump version to 1.5.1 so that versions will match (#126)

jacquescaocb committed:
Add ConstructionPreprocessOperations API endpoint (#127)

This endpoint provides fallback parsing for high-level transaction
construction operations when local OperationSelector doesn't support
certain construct operations.

Features:
- Input format matches TransactionConstructOpInput from chainstdio
- Output format matches ParseTransactionConstructOp return values
- Optional API (servers can return 501 Not Implemented)
- Full OpenAPI specification with request/response schemas
- Maintains compatibility with existing operation parsing workflows

Endpoint: POST /construction/preprocess_operations
Request: ConstructionPreprocessOperationsRequest
Response: ConstructionPreprocessOperationsResponse

jacquescaocb committed:
feat: All account balances (#128)

* Add AllAccountBalances API and fix URL naming consistency

This commit introduces two key enhancements:

1. AllAccountBalances API:
   - Added POST /account/all-balances endpoint
   - Consolidates multiple sub-account balance queries into single request
   - Ensures consistent block context for all balances
   - Added AllAccountBalancesRequest, AllAccountBalancesResponse, and AccountBalanceWithSubAccount schemas
   - Removed root-level metadata from AllAccountBalancesResponse (metadata belongs to individual sub-accounts)

2. URL Naming Consistency:
   - Changed /construction/preprocess_operations to /construction/preprocess-operations
   - Maintains kebab-case convention across all Rosetta endpoints
   - Updated both endpoint path and description references

Benefits:
- Reduced network latency (single request vs multiple)
- Improved data consistency (same block context)
- Backward compatible (existing /account/balance unchanged)
- Optional implementation for Rosetta servers

Files modified:
- api.yaml: Added new endpoint and schemas, fixed URL naming
- api.json: Regenerated specification bundle

* Remove SubAccountIdentifier field from AccountBalanceWithSubAccount schema

- Remove sub_account_identifier field from AllAccountBalancesResponse
- Update descriptions to reflect that balance identification relies on balance_type and metadata
- Simplifies API by removing redundant field that is rarely used in practice

Wiezzel committed:
Add ml_dsa_65 (FIPS 204) to CurveType and SignatureType (#129)

Introduces `ml_dsa_65` as a new value for both `CurveType` and
`SignatureType`. ML-DSA-65 is the medium-strength FIPS 204 module-lattice
digital-signature variant (NIST Security Category 3): 1952-byte public
key, 3309-byte signature.

The `CurveType` field is also annotated to note that the value may now
identify a post-quantum scheme that is not strictly an elliptic curve.
The enum and field name are kept unchanged for backwards compatibility;
clients that do not recognise `ml_dsa_65` should reject it as they would
any unknown enum value.

`api.json` is regenerated via `make gen` from `models/CurveType.yaml`
and `models/SignatureType.yaml`.

Motivation: the NEAR Protocol is adding ML-DSA-65 as a third
transaction-signature scheme alongside ed25519 and secp256k1, and its
Mesh / Rosetta RPC implementation needs a way to represent
post-quantum public keys and signatures at the API boundary. Other
networks adopting post-quantum signatures are expected to need the same
discriminator value.
